### DISCUSSION TITLE
Adding ADBC Driver as engine option

### DISCUSSION THREAD

**Discussion by MMCMA** [sqlalchemy/sqlalchemy#11743] [Ideas]
### Describe the use case

I've recently came across [Arrow ADBC](https://arrow.apache.org/adbc/current/index.html) which seems to offer great performance improvements compared to other DBAPIs (depending on the use case of course)

### Databases / Backends / Drivers targeted

It is database agnostic, but more details can be found [here](https://arrow.apache.org/blog/2023/01/05/introducing-arrow-adbc/)

### Example Use

```
import adbc_driver_postgresql.dbapi

uri = "postgresql://localhost:5432/postgres?user=postgres&password=password"
with adbc_driver_postgresql.dbapi.connect(uri) as conn:
    with conn.cursor() as cur:
        cur.execute("SELECT * FROM customer")
        table = cur.fetch_arrow_table()
```

### Additional context

I initially added a question on the ADBC project, see https://github.com/apache/arrow-adbc/issues/2079 for more details.

**Comment by CaselIT** [comment]
Hi,

I think it could be possible to implement a dialect that uses this driver, but it would likely be in an external library, and be packaged like a dialect.

Note that I'm not sure the use case fits with sqlalchemy, since adbc seems to target column wise operations, so without using raw cursors you would loose most of its advantages.

moving to discussion

**Comment by MMCMA** [comment]
I am afraid my knowledge of low-level arrow ADBC is very limited. Feel free to comment also here on the [issue](https://github.com/apache/arrow-adbc/issues/2079)
